pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,300 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from contextlib import suppress
+from typing import Any, List, Optional, Tuple
+
+from deprecate import void
+from torch.optim import Optimizer
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops.base import Loop
+from pytorch_lightning.loops.training_epoch_loop import TrainingEpochLoop
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.trainer.supporters import TensorRunningAccum
+from pytorch_lightning.utilities import rank_zero_info
+
+log = logging.getLogger(__name__)
+
+
+class FitLoop(Loop):
+    """This Loop iterates over the epochs to run the training
+
+    Args:
+        min_epochs: The minimum number of epochs
+        max_epochs: The maximum number of epochs
+        min_steps: The minimum number of steps
+        max_steps: The maximum number of epoch
+
+    .. note::
+        If neither the minimum epochs nor steps are specified the minimum number of epochs is set to 1
+        and if neither the maximum steps nor epochs are specified, the maximum epochs are set to 1000.
+    """
+
+    def __init__(
+        self,
+        min_epochs: Optional[int] = None,
+        max_epochs: Optional[int] = None,
+        min_steps: Optional[int] = None,
+        max_steps: Optional[int] = None
+    ):
+        super().__init__()
+        # If neither max_epochs or max_steps is set, then use existing default of max_epochs = 1000
+        self.max_epochs = 1000 if (max_epochs is None and max_steps is None) else max_epochs
+        # If neither min_epochs or min_steps is set, then use existing default of min_epochs = 1
+        self.min_epochs = 1 if (min_epochs is None and min_steps is None) else min_epochs
```

Comment:
Why set these here and then pass them to `training_loop`? Shouldn't this logic be there and just have a property here?

Reply:
so the epoch related mins maxes are set here because fit loop manages epochs so it is the responsibility of it to set the values.

and the step related mins and maxes are managed by the other loop. 
when we refine the instantiation of the loops we will with connect() etc. we will aim to separate it even more cleaner!